### PR TITLE
fix(tab-bar): Move activateTab to adapter

### PR DESCRIPTION
**Pull request by bonniezhou**
Move the job of calling `activateTab` into an adapter method instead of directly in the foundation's handler methods. In the React wrapper, handler methods should only update props. `activateTab` will then be called in the `componentDidUpdate` lifecycle method. 

Also rename `getActiveTabIndex` to `getPreviousActiveTabIndex` to disambiguate. React wrapper stores the previous active index in state and returns it in this adapter method.

### CODE REVIEW

**Comment by kfranqueiro** [comment]
This seems super-weird from a MDC Web core perspective. We're taking what used to be one foundation method calling another, and changing it to make the foundation call the adapter to immediately call into the foundation again. Typically if anything we've been trying lately to _avoid_ "hollow" loops between component->foundation->adapter, and this seems like an even smaller loop than those.

You said that "handler methods should only update props"... can you explain or point to what exactly is happening right now and why it's a problem?

**Comment by bonniezhou** [comment]
This is different from the component->foundation->adapter loops, it's more like foundation->adapter->foundation. The main reason for doing this is to separate the foundation's `activateTab` method from the foundation's handler methods.

The problem right now is that in React, the user expects to have full control over the tab bar's active index. Say the user keeps this `activeIndex` in some state variable, and passes it as a prop to the `TabBar`. Every time the `TabBar` detects that props have changed, it will call the foundation's `activateTab` (through the `componentDidUpdate` lifecycle method).

This means that tab bar's click/keydown handlers should be calling a callback that updates the user's `activeIndex` state, not calling `activateTab` directly.

**Comment by kfranqueiro** [comment]
I'm trying to think of a way to formalize this so we can understand why this is done and know when we need to do it (vs. when it's not necessary). It's hard for me to pin this down generically - is it something like "if a foundation's handler needs to change state, it needs to go through an adapter"? I guess that would conform to e.g. Switch's `handleChange` applying CSS classes via the adapter...

**Comment by bonniezhou** [comment]
I think "if a foundation's handler needs to change state, it needs to go through an adapter" makes sense.

Yeah foreseeing "state" is tricky. If it helps, I found an example of something that does _not_ hold state: the foundation's `scrollIntoView` method. It's similar to `activateTab` in that it's something that the user should be able to directly manipulate (via MDCW's `MDCTabBar.scrollIntoView` method or MDCR's `indexInView` prop), but it doesn't manage something that changes internally and needs to be propagated back up to the user.

File: packages/mdc-tab-bar/foundation.js [hunk 1]
```diff
@@ -106,7 +106,8 @@ class MDCTabBarFoundation extends MDCFoundation {
   }
 
   init() {
-    const activeIndex = this.adapter_.getActiveTabIndex();
+    // Upon init, the active tab index is the same as the previous active tab index
+    const activeIndex = this.adapter_.getPreviousActiveTabIndex();
```
Comment (patrickrodee) [inline packages/mdc-tab-bar/foundation.js]: This one line feels weird. All other cases seem like improvements.

**Comment by kfranqueiro** [comment]
Remember `BREAKING CHANGE:` for renamed and added adapter APIs

**Pull request merged**